### ISSUE TITLE
Command to check that all symbols come from declared packages

### ISSUE THREAD

**Issue by marcosh** [purescript/spago#598]
As far as I understand, with Spago and Purescript it is possible to use data types and functions coming from transitive dependencies (dependencies of declared dependencies).

This may cause problems if my declared dependency decides to remove one of its dependency which I'm using but not declaring.

To avoid such issues, it would be nice to have in Spago a command which checks that every function and every data type used in the project comes from a package declared as a dependency in `spago.dhall`. Moreover, the same command could check whether some declared dependency is actually not used by the project and could be removed.

**Comment by f-f** [comment]
Yes, I think we should run this check. However, right now it's not really feasible to do it, as Spago would have to parse module imports, as we should really not do that. So in order to enable this (and other things), I implemented `purs graph` in the compiler (see ).

This will ship in `purs@v0.14.0`, and will allow us to get a mapping of what every module imports, together with the path to every module in the build graph. Example of running it on the default Spago project (command `purs graph src/Main.purs '.spago/**/*.purs' | jq .`):
```json
{
  "Data.BooleanAlgebra": {
    "path": ".spago/prelude/v4.1.1/src/Data/BooleanAlgebra.purs",
    "depends": [
      "Data.HeytingAlgebra",
      "Data.Symbol",
      "Data.Unit",
      "Data.Show",
      "Record.Unsafe",
      "Type.Data.RowList",
      "Type.Data.Row"
    ]
  },
  "Data.Ring": {
    "path": ".spago/prelude/v4.1.1/src/Data/Ring.purs",
    "depends": [
      "Data.Semiring",
      "Data.Symbol",
      "Data.Unit",
      "Data.Show",
      "Record.Unsafe",
      "Type.Data.RowList",
      "Type.Data.Row"
    ]
  },
  "Effect.Uncurried": {
    "path": ".spago/effect/v2.0.1/src/Effect/Uncurried.purs",
    "depends": [
      "Effect",
      "Prelude",
      "Control.Applicative",
      "Control.Apply",
      "Data.Functor",
      "Data.Function",
      "Control.Category",
      "Control.Semigroupoid",
      "Data.Boolean",
      "Data.Ord",
      "Data.Eq",
      "Data.HeytingAlgebra",
      "Data.Symbol",
      "Data.Unit",
      "Data.Show",
      "Record.Unsafe",
      "Type.Data.RowList",
      "Type.Data.Row",
      "Data.Void",
      "Data.Ordering",
      "Data.Semigroup",
      "Data.Ring",
      "Data.Semiring",
      "Control.Bind",
      "Control.Monad",
      "Data.BooleanAlgebra",
      "Data.Bounded",
      "Data.CommutativeRing",
      "Data.DivisionRing",
      "Data.EuclideanRing",
      "Data.Field",
      "Data.Monoid",
      "Data.NaturalTransformation"
    ]
  },
  "Effect": {
    "path": ".spago/effect/v2.0.1/src/Effect.purs",
    "depends": [
      "Prelude",
      "Control.Applicative",
      "Control.Apply",
      "Data.Functor",
      "Data.Function",
      "Control.Category",
      "Control.Semigroupoid",
      "Data.Boolean",
      "Data.Ord",
      "Data.Eq",
      "Data.HeytingAlgebra",
      "Data.Symbol",
      "Data.Unit",
      "Data.Show",
      "Record.Unsafe",
      "Type.Data.RowList",
      "Type.Data.Row",
      "Data.Void",
      "Data.Ordering",
      "Data.Semigroup",
      "Data.Ring",
      "Data.Semiring",
      "Control.Bind",
      "Control.Monad",
      "Data.BooleanAlgebra",
      "Data.Bounded",
      "Data.CommutativeRing",
      "Data.DivisionRing",
      "Data.EuclideanRing",
      "Data.Field",
      "Data.Monoid",
      "Data.NaturalTransformation"
    ]
  },
  "Data.Ord": {
    "path": ".spago/prelude/v4.1.1/src/Data/Ord.purs",
    "depends": [
      "Data.Eq",
      "Data.HeytingAlgebra",
      "Data.Symbol",
      "Data.Unit",
      "Data.Show",
      "Record.Unsafe",
      "Type.Data.RowList",
      "Type.Data.Row",
      "Data.Void",
      "Data.Ordering",
      "Data.Semigroup",
      "Data.Ring",
      "Data.Semiring"
    ]
  },
  "Type.Data.Row": {
    "path": ".spago/prelude/v4.1.1/src/Type/Data/Row.purs",
    "depends": []
  },
  "Data.Monoid.Dual": {
    "path": ".spago/prelude/v4.1.1/src/Data/Monoid/Dual.purs",
    "depends": [
      "Prelude",
      "Control.Applicative",
      "Control.Apply",
      "Data.Functor",
      "Data.Function",
      "Control.Category",
      "Control.Semigroupoid",
      "Data.Boolean",
      "Data.Ord",
      "Data.Eq",
      "Data.HeytingAlgebra",
      "Data.Symbol",
      "Data.Unit",
      "Data.Show",
      "Record.Unsafe",
      "Type.Data.RowList",
      "Type.Data.Row",
      "Data.Void",
      "Data.Ordering",
      "Data.Semigroup",
      "Data.Ring",
      "Data.Semiring",
      "Control.Bind",
      "Control.Monad",
      "Data.BooleanAlgebra",
      "Data.Bounded",
      "Data.CommutativeRing",
      "Data.DivisionRing",
      "Data.EuclideanRing",
      "Data.Field",
      "Data.Monoid",
      "Data.NaturalTransformation"
    ]
  },
  "Data.Boolean": {
    "path": ".spago/prelude/v4.1.1/src/Data/Boolean.purs",
    "depends": []
  },
  "Data.Bounded": {
    "path": ".spago/prelude/v4.1.1/src/Data/Bounded.purs",
    "depends": [
      "Data.Ord",
      "Data.Eq",
      "Data.HeytingAlgebra",
      "Data.Symbol",
      "Data.Unit",
      "Data.Show",
      "Record.Unsafe",
      "Type.Data.RowList",
      "Type.Data.Row",
      "Data.Void",
      "Data.Ordering",
      "Data.Semigroup",
      "Data.Ring",
      "Data.Semiring"
    ]
  },
  "Data.Show": {
    "path": ".spago/prelude/v4.1.1/src/Data/Show.purs",
    "depends": [
      "Data.Symbol",
      "Record.Unsafe",
      "Type.Data.RowList"
    ]
  },
  "Control.Apply": {
    "path": ".spago/prelude/v4.1.1/src/Control/Apply.purs",
    "depends": [
      "Data.Functor",
      "Data.Function",
      "Control.Category",
      "Control.Semigroupoid",
      "Data.Boolean",
      "Data.Ord",
      "Data.Eq",
      "Data.HeytingAlgebra",
      "Data.Symbol",
      "Data.Unit",
      "Data.Show",
      "Record.Unsafe",
      "Type.Data.RowList",
      "Type.Data.Row",
      "Data.Void",
      "Data.Ordering",
      "Data.Semigroup",
      "Data.Ring",
      "Data.Semiring"
    ]
  },
  "Control.Monad": {
    "path": ".spago/prelude/v4.1.1/src/Control/Monad.purs",
    "depends": [
      "Control.Applicative",
      "Control.Apply",
      "Data.Functor",
      "Data.Function",
      "Control.Category",
      "Control.Semigroupoid",
      "Data.Boolean",
      "Data.Ord",
      "Data.Eq",
      "Data.HeytingAlgebra",
      "Data.Symbol",
      "Data.Unit",
      "Data.Show",
      "Record.Unsafe",
      "Type.Data.RowList",
      "Type.Data.Row",
      "Data.Void",
      "Data.Ordering",
      "Data.Semigroup",
      "Data.Ring",
      "Data.Semiring",
      "Control.Bind"
    ]
  },
  "Data.Monoid": {
    "path": ".spago/prelude/v4.1.1/src/Data/Monoid.purs",
    "depends": [
      "Data.Boolean",
      "Data.Eq",
      "Data.HeytingAlgebra",
      "Data.Symbol",
      "Data.Unit",
      "Data.Show",
      "Record.Unsafe",
      "Type.Data.RowList",
      "Type.Data.Row",
      "Data.Void",
      "Data.EuclideanRing",
      "Data.BooleanAlgebra",
      "Data.CommutativeRing",
      "Data.Ring",
      "Data.Semiring",
      "Data.Ord",
      "Data.Ordering",
      "Data.Semigroup"
    ]
  },
  "Control.Bind": {
    "path": ".spago/prelude/v4.1.1/src/Control/Bind.purs",
    "depends": [
      "Control.Applicative",
      "Control.Apply",
      "Data.Functor",
      "Data.Function",
      "Control.Category",
      "Control.Semigroupoid",
      "Data.Boolean",
      "Data.Ord",
      "Data.Eq",
      "Data.HeytingAlgebra",
      "Data.Symbol",
      "Data.Unit",
      "Data.Show",
      "Record.Unsafe",
      "Type.Data.RowList",
      "Type.Data.Row",
      "Data.Void",
      "Data.Ordering",
      "Data.Semigroup",
      "Data.Ring",
      "Data.Semiring"
    ]
  },
  "Data.Monoid.Additive": {
    "path": ".spago/prelude/v4.1.1/src/Data/Monoid/Additive.purs",
    "depends": [
      "Prelude",
      "Control.Applicative",
      "Control.Apply",
      "Data.Functor",
      "Data.Function",
      "Control.Category",
      "Control.Semigroupoid",
      "Data.Boolean",
      "Data.Ord",
      "Data.Eq",
      "Data.HeytingAlgebra",
      "Data.Symbol",
      "Data.Unit",
      "Data.Show",
      "Record.Unsafe",
      "Type.Data.RowList",
      "Type.Data.Row",
      "Data.Void",
      "Data.Ordering",
      "Data.Semigroup",
      "Data.Ring",
      "Data.Semiring",
      "Control.Bind",
      "Control.Monad",
      "Data.BooleanAlgebra",
      "Data.Bounded",
      "Data.CommutativeRing",
      "Data.DivisionRing",
      "Data.EuclideanRing",
      "Data.Field",
      "Data.Monoid",
      "Data.NaturalTransformation"
    ]
  },
  "Data.Symbol": {
    "path": ".spago/prelude/v4.1.1/src/Data/Symbol.purs",
    "depends": []
  },
  "Data.HeytingAlgebra": {
    "path": ".spago/prelude/v4.1.1/src/Data/HeytingAlgebra.purs",
    "depends": [
      "Data.Symbol",
      "Data.Unit",
      "Data.Show",
      "Record.Unsafe",
      "Type.Data.RowList",
      "Type.Data.Row"
    ]
  },
  "Type.Data.RowList": {
    "path": ".spago/prelude/v4.1.1/src/Type/Data/RowList.purs",
    "depends": []
  },
  "Data.Ord.Unsafe": {
    "path": ".spago/prelude/v4.1.1/src/Data/Ord/Unsafe.purs",
    "depends": [
      "Data.Ordering",
      "Data.Eq",
      "Data.HeytingAlgebra",
      "Data.Symbol",
      "Data.Unit",
      "Data.Show",
      "Record.Unsafe",
      "Type.Data.RowList",
      "Type.Data.Row",
      "Data.Void",
      "Data.Semigroup"
    ]
  },
  "Data.Semigroup": {
    "path": ".spago/prelude/v4.1.1/src/Data/Semigroup.purs",
    "depends": [
      "Data.Symbol",
      "Data.Unit",
      "Data.Show",
      "Record.Unsafe",
      "Type.Data.RowList",
      "Data.Void"
    ]
  },
  "Test.Main": {
    "path": ".spago/prelude/v4.1.1/test/Test/Main.purs",
    "depends": [
      "Prelude",
      "Control.Applicative",
      "Control.Apply",
      "Data.Functor",
      "Data.Function",
      "Control.Category",
      "Control.Semigroupoid",
      "Data.Boolean",
      "Data.Ord",
      "Data.Eq",
      "Data.HeytingAlgebra",
      "Data.Symbol",
      "Data.Unit",
      "Data.Show",
      "Record.Unsafe",
      "Type.Data.RowList",
      "Type.Data.Row",
      "Data.Void",
      "Data.Ordering",
      "Data.Semigroup",
      "Data.Ring",
      "Data.Semiring",
      "Control.Bind",
      "Control.Monad",
      "Data.BooleanAlgebra",
      "Data.Bounded",
      "Data.CommutativeRing",
      "Data.DivisionRing",
      "Data.EuclideanRing",
      "Data.Field",
      "Data.Monoid",
      "Data.NaturalTransformation",
      "PSCI.Support",
      "Effect",
      "Effect.Console"
    ]
  },
  "Main": {
    "path": "src/Main.purs",
    "depends": [
      "Prelude",
      "Control.Applicative",
      "Control.Apply",
      "Data.Functor",
      "Data.Function",
      "Control.Category",
      "Control.Semigroupoid",
      "Data.Boolean",
      "Data.Ord",
      "Data.Eq",
      "Data.HeytingAlgebra",
      "Data.Symbol",
      "Data.Unit",
      "Data.Show",
      "Record.Unsafe",
      "Type.Data.RowList",
      "Type.Data.Row",
      "Data.Void",
      "Data.Ordering",
      "Data.Semigroup",
      "Data.Ring",
      "Data.Semiring",
      "Control.Bind",
      "Control.Monad",
      "Data.BooleanAlgebra",
      "Data.Bounded",
      "Data.CommutativeRing",
      "Data.DivisionRing",
      "Data.EuclideanRing",
      "Data.Field",
      "Data.Monoid",
      "Data.NaturalTransformation",
      "Effect",
      "Effect.Console"
    ]
  },
  "Effect.Unsafe": {
    "path": ".spago/effect/v2.0.1/src/Effect/Unsafe.purs",
    "depends": [
      "Effect",
      "Prelude",
      "Control.Applicative",
      "Control.Apply",
      "Data.Functor",
      "Data.Function",
      "Control.Category",
      "Control.Semigroupoid",
      "Data.Boolean",
      "Data.Ord",
      "Data.Eq",
      "Data.HeytingAlgebra",
      "Data.Symbol",
      "Data.Unit",
      "Data.Show",
      "Record.Unsafe",
      "Type.Data.RowList",
      "Type.Data.Row",
      "Data.Void",
      "Data.Ordering",
      "Data.Semigroup",
      "Data.Ring",
      "Data.Semiring",
      "Control.Bind",
      "Control.Monad",
      "Data.BooleanAlgebra",
      "Data.Bounded",
      "Data.CommutativeRing",
      "Data.DivisionRing",
      "Data.EuclideanRing",
      "Data.Field",
      "Data.Monoid",
      "Data.NaturalTransformation"
    ]
  },
  "Data.Semiring": {
    "path": ".spago/prelude/v4.1.1/src/Data/Semiring.purs",
    "depends": [
      "Data.Symbol",
      "Data.Unit",
      "Data.Show",
      "Record.Unsafe",
      "Type.Data.RowList",
      "Type.Data.Row"
    ]
  },
  "Effect.Console": {
    "path": ".spago/console/v4.4.0/src/Effect/Console.purs",
    "depends": [
      "Effect",
      "Prelude",
      "Control.Applicative",
      "Control.Apply",
      "Data.Functor",
      "Data.Function",
      "Control.Category",
      "Control.Semigroupoid",
      "Data.Boolean",
      "Data.Ord",
      "Data.Eq",
      "Data.HeytingAlgebra",
      "Data.Symbol",
      "Data.Unit",
      "Data.Show",
      "Record.Unsafe",
      "Type.Data.RowList",
      "Type.Data.Row",
      "Data.Void",
      "Data.Ordering",
      "Data.Semigroup",
      "Data.Ring",
      "Data.Semiring",
      "Control.Bind",
      "Control.Monad",
      "Data.BooleanAlgebra",
      "Data.Bounded",
      "Data.CommutativeRing",
      "Data.DivisionRing",
      "Data.EuclideanRing",
      "Data.Field",
      "Data.Monoid",
      "Data.NaturalTransformation"
    ]
  },
  "Data.CommutativeRing": {
    "path": ".spago/prelude/v4.1.1/src/Data/CommutativeRing.purs",
    "depends": [
      "Data.Ring",
      "Data.Semiring",
      "Data.Symbol",
      "Data.Unit",
      "Data.Show",
      "Record.Unsafe",
      "Type.Data.RowList",
      "Type.Data.Row"
    ]
  },
  "Data.NaturalTransformation": {
    "path": ".spago/prelude/v4.1.1/src/Data/NaturalTransformation.purs",
    "depends": []
  },
  "Data.Monoid.Conj": {
    "path": ".spago/prelude/v4.1.1/src/Data/Monoid/Conj.purs",
    "depends": [
      "Prelude",
      "Control.Applicative",
      "Control.Apply",
      "Data.Functor",
      "Data.Function",
      "Control.Category",
      "Control.Semigroupoid",
      "Data.Boolean",
      "Data.Ord",
      "Data.Eq",
      "Data.HeytingAlgebra",
      "Data.Symbol",
      "Data.Unit",
      "Data.Show",
      "Record.Unsafe",
      "Type.Data.RowList",
      "Type.Data.Row",
      "Data.Void",
      "Data.Ordering",
      "Data.Semigroup",
      "Data.Ring",
      "Data.Semiring",
      "Control.Bind",
      "Control.Monad",
      "Data.BooleanAlgebra",
      "Data.Bounded",
      "Data.CommutativeRing",
      "Data.DivisionRing",
      "Data.EuclideanRing",
      "Data.Field",
      "Data.Monoid",
      "Data.NaturalTransformation"
    ]
  },
  "Data.Functor": {
    "path": ".spago/prelude/v4.1.1/src/Data/Functor.purs",
    "depends": [
      "Data.Function",
      "Control.Category",
      "Control.Semigroupoid",
      "Data.Boolean",
      "Data.Ord",
      "Data.Eq",
      "Data.HeytingAlgebra",
      "Data.Symbol",
      "Data.Unit",
      "Data.Show",
      "Record.Unsafe",
      "Type.Data.RowList",
      "Type.Data.Row",
      "Data.Void",
      "Data.Ordering",
      "Data.Semigroup",
      "Data.Ring",
      "Data.Semiring"
    ]
  },
  "Control.Category": {
    "path": ".spago/prelude/v4.1.1/src/Control/Category.purs",
    "depends": [
      "Control.Semigroupoid"
    ]
  },
  "Data.Function": {
    "path": ".spago/prelude/v4.1.1/src/Data/Function.purs",
    "depends": [
      "Control.Category",
      "Control.Semigroupoid",
      "Data.Boolean",
      "Data.Ord",
      "Data.Eq",
      "Data.HeytingAlgebra",
      "Data.Symbol",
      "Data.Unit",
      "Data.Show",
      "Record.Unsafe",
      "Type.Data.RowList",
      "Type.Data.Row",
      "Data.Void",
      "Data.Ordering",
      "Data.Semigroup",
      "Data.Ring",
      "Data.Semiring"
    ]
  },
  "Data.Field": {
    "path": ".spago/prelude/v4.1.1/src/Data/Field.purs",
    "depends": [
      "Data.DivisionRing",
      "Data.EuclideanRing",
      "Data.BooleanAlgebra",
      "Data.HeytingAlgebra",
      "Data.Symbol",
      "Data.Unit",
      "Data.Show",
      "Record.Unsafe",
      "Type.Data.RowList",
      "Type.Data.Row",
      "Data.CommutativeRing",
      "Data.Ring",
      "Data.Semiring",
      "Data.Eq",
      "Data.Void"
    ]
  },
  "Data.EuclideanRing": {
    "path": ".spago/prelude/v4.1.1/src/Data/EuclideanRing.purs",
    "depends": [
      "Data.BooleanAlgebra",
      "Data.HeytingAlgebra",
      "Data.Symbol",
      "Data.Unit",
      "Data.Show",
      "Record.Unsafe",
      "Type.Data.RowList",
      "Type.Data.Row",
      "Data.CommutativeRing",
      "Data.Ring",
      "Data.Semiring",
      "Data.Eq",
      "Data.Void"
    ]
  },
  "Data.Semigroup.Last": {
    "path": ".spago/prelude/v4.1.1/src/Data/Semigroup/Last.purs",
    "depends": [
      "Prelude",
      "Control.Applicative",
      "Control.Apply",
      "Data.Functor",
      "Data.Function",
      "Control.Category",
      "Control.Semigroupoid",
      "Data.Boolean",
      "Data.Ord",
      "Data.Eq",
      "Data.HeytingAlgebra",
      "Data.Symbol",
      "Data.Unit",
      "Data.Show",
      "Record.Unsafe",
      "Type.Data.RowList",
      "Type.Data.Row",
      "Data.Void",
      "Data.Ordering",
      "Data.Semigroup",
      "Data.Ring",
      "Data.Semiring",
      "Control.Bind",
      "Control.Monad",
      "Data.BooleanAlgebra",
      "Data.Bounded",
      "Data.CommutativeRing",
      "Data.DivisionRing",
      "Data.EuclideanRing",
      "Data.Field",
      "Data.Monoid",
      "Data.NaturalTransformation"
    ]
  },
  "Data.Semigroup.First": {
    "path": ".spago/prelude/v4.1.1/src/Data/Semigroup/First.purs",
    "depends": [
      "Prelude",
      "Control.Applicative",
      "Control.Apply",
      "Data.Functor",
      "Data.Function",
      "Control.Category",
      "Control.Semigroupoid",
      "Data.Boolean",
      "Data.Ord",
      "Data.Eq",
      "Data.HeytingAlgebra",
      "Data.Symbol",
      "Data.Unit",
      "Data.Show",
      "Record.Unsafe",
      "Type.Data.RowList",
      "Type.Data.Row",
      "Data.Void",
      "Data.Ordering",
      "Data.Semigroup",
      "Data.Ring",
      "Data.Semiring",
      "Control.Bind",
      "Control.Monad",
      "Data.BooleanAlgebra",
      "Data.Bounded",
      "Data.CommutativeRing",
      "Data.DivisionRing",
      "Data.EuclideanRing",
      "Data.Field",
      "Data.Monoid",
      "Data.NaturalTransformation"
    ]
  },
  "Prelude": {
    "path": ".spago/prelude/v4.1.1/src/Prelude.purs",
    "depends": [
      "Control.Applicative",
      "Control.Apply",
      "Data.Functor",
      "Data.Function",
      "Control.Category",
      "Control.Semigroupoid",
      "Data.Boolean",
      "Data.Ord",
      "Data.Eq",
      "Data.HeytingAlgebra",
      "Data.Symbol",
      "Data.Unit",
      "Data.Show",
      "Record.Unsafe",
      "Type.Data.RowList",
      "Type.Data.Row",
      "Data.Void",
      "Data.Ordering",
      "Data.Semigroup",
      "Data.Ring",
      "Data.Semiring",
      "Control.Bind",
      "Control.Monad",
      "Data.BooleanAlgebra",
      "Data.Bounded",
      "Data.CommutativeRing",
      "Data.DivisionRing",
      "Data.EuclideanRing",
      "Data.Field",
      "Data.Monoid",
      "Data.NaturalTransformation"
    ]
  },
  "Effect.Class": {
    "path": ".spago/effect/v2.0.1/src/Effect/Class.purs",
    "depends": [
      "Control.Category",
      "Control.Semigroupoid",
      "Control.Monad",
      "Control.Applicative",
      "Control.Apply",
      "Data.Functor",
      "Data.Function",
      "Data.Boolean",
      "Data.Ord",
      "Data.Eq",
      "Data.HeytingAlgebra",
      "Data.Symbol",
      "Data.Unit",
      "Data.Show",
      "Record.Unsafe",
      "Type.Data.RowList",
      "Type.Data.Row",
      "Data.Void",
      "Data.Ordering",
      "Data.Semigroup",
      "Data.Ring",
      "Data.Semiring",
      "Control.Bind",
      "Effect",
      "Prelude",
      "Data.BooleanAlgebra",
      "Data.Bounded",
      "Data.CommutativeRing",
      "Data.DivisionRing",
      "Data.EuclideanRing",
      "Data.Field",
      "Data.Monoid",
      "Data.NaturalTransformation"
    ]
  },
  "Data.Eq": {
    "path": ".spago/prelude/v4.1.1/src/Data/Eq.purs",
    "depends": [
      "Data.HeytingAlgebra",
      "Data.Symbol",
      "Data.Unit",
      "Data.Show",
      "Record.Unsafe",
      "Type.Data.RowList",
      "Type.Data.Row",
      "Data.Void"
    ]
  },
  "Data.Monoid.Disj": {
    "path": ".spago/prelude/v4.1.1/src/Data/Monoid/Disj.purs",
    "depends": [
      "Prelude",
      "Control.Applicative",
      "Control.Apply",
      "Data.Functor",
      "Data.Function",
      "Control.Category",
      "Control.Semigroupoid",
      "Data.Boolean",
      "Data.Ord",
      "Data.Eq",
      "Data.HeytingAlgebra",
      "Data.Symbol",
      "Data.Unit",
      "Data.Show",
      "Record.Unsafe",
      "Type.Data.RowList",
      "Type.Data.Row",
      "Data.Void",
      "Data.Ordering",
      "Data.Semigroup",
      "Data.Ring",
      "Data.Semiring",
      "Control.Bind",
      "Control.Monad",
      "Data.BooleanAlgebra",
      "Data.Bounded",
      "Data.CommutativeRing",
      "Data.DivisionRing",
      "Data.EuclideanRing",
      "Data.Field",
      "Data.Monoid",
      "Data.NaturalTransformation"
    ]
  },
  "Data.Void": {
    "path": ".spago/prelude/v4.1.1/src/Data/Void.purs",
    "depends": [
      "Data.Show",
      "Data.Symbol",
      "Record.Unsafe",
      "Type.Data.RowList"
    ]
  },
  "Data.DivisionRing": {
    "path": ".spago/prelude/v4.1.1/src/Data/DivisionRing.purs",
    "depends": [
      "Data.EuclideanRing",
      "Data.BooleanAlgebra",
      "Data.HeytingAlgebra",
      "Data.Symbol",
      "Data.Unit",
      "Data.Show",
      "Record.Unsafe",
      "Type.Data.RowList",
      "Type.Data.Row",
      "Data.CommutativeRing",
      "Data.Ring",
      "Data.Semiring",
      "Data.Eq",
      "Data.Void"
    ]
  },
  "PSCI.Support": {
    "path": ".spago/psci-support/v4.0.0/src/PSCI/Support.purs",
    "depends": [
      "Prelude",
      "Control.Applicative",
      "Control.Apply",
      "Data.Functor",
      "Data.Function",
      "Control.Category",
      "Control.Semigroupoid",
      "Data.Boolean",
      "Data.Ord",
      "Data.Eq",
      "Data.HeytingAlgebra",
      "Data.Symbol",
      "Data.Unit",
      "Data.Show",
      "Record.Unsafe",
      "Type.Data.RowList",
      "Type.Data.Row",
      "Data.Void",
      "Data.Ordering",
      "Data.Semigroup",
      "Data.Ring",
      "Data.Semiring",
      "Control.Bind",
      "Control.Monad",
      "Data.BooleanAlgebra",
      "Data.Bounded",
      "Data.CommutativeRing",
      "Data.DivisionRing",
      "Data.EuclideanRing",
      "Data.Field",
      "Data.Monoid",
      "Data.NaturalTransformation",
      "Effect",
      "Effect.Console"
    ]
  },
  "Data.Unit": {
    "path": ".spago/prelude/v4.1.1/src/Data/Unit.purs",
    "depends": [
      "Data.Show",
      "Data.Symbol",
      "Record.Unsafe",
      "Type.Data.RowList"
    ]
  },
  "Data.Ordering": {
    "path": ".spago/prelude/v4.1.1/src/Data/Ordering.purs",
    "depends": [
      "Data.Eq",
      "Data.HeytingAlgebra",
      "Data.Symbol",
      "Data.Unit",
      "Data.Show",
      "Record.Unsafe",
      "Type.Data.RowList",
      "Type.Data.Row",
      "Data.Void",
      "Data.Semigroup"
    ]
  },
  "Data.Monoid.Multiplicative": {
    "path": ".spago/prelude/v4.1.1/src/Data/Monoid/Multiplicative.purs",
    "depends": [
      "Prelude",
      "Control.Applicative",
      "Control.Apply",
      "Data.Functor",
      "Data.Function",
      "Control.Category",
      "Control.Semigroupoid",
      "Data.Boolean",
      "Data.Ord",
      "Data.Eq",
      "Data.HeytingAlgebra",
      "Data.Symbol",
      "Data.Unit",
      "Data.Show",
      "Record.Unsafe",
      "Type.Data.RowList",
      "Type.Data.Row",
      "Data.Void",
      "Data.Ordering",
      "Data.Semigroup",
      "Data.Ring",
      "Data.Semiring",
      "Control.Bind",
      "Control.Monad",
      "Data.BooleanAlgebra",
      "Data.Bounded",
      "Data.CommutativeRing",
      "Data.DivisionRing",
      "Data.EuclideanRing",
      "Data.Field",
      "Data.Monoid",
      "Data.NaturalTransformation"
    ]
  },
  "Control.Semigroupoid": {
    "path": ".spago/prelude/v4.1.1/src/Control/Semigroupoid.purs",
    "depends": []
  },
  "Effect.Class.Console": {
    "path": ".spago/console/v4.4.0/src/Effect/Class/Console.purs",
    "depends": [
      "Data.Function",
      "Control.Category",
      "Control.Semigroupoid",
      "Data.Boolean",
      "Data.Ord",
      "Data.Eq",
      "Data.HeytingAlgebra",
      "Data.Symbol",
      "Data.Unit",
      "Data.Show",
      "Record.Unsafe",
      "Type.Data.RowList",
      "Type.Data.Row",
      "Data.Void",
      "Data.Ordering",
      "Data.Semigroup",
      "Data.Ring",
      "Data.Semiring",
      "Effect.Class",
      "Control.Monad",
      "Control.Applicative",
      "Control.Apply",
      "Data.Functor",
      "Control.Bind",
      "Effect",
      "Prelude",
      "Data.BooleanAlgebra",
      "Data.Bounded",
      "Data.CommutativeRing",
      "Data.DivisionRing",
      "Data.EuclideanRing",
      "Data.Field",
      "Data.Monoid",
      "Data.NaturalTransformation",
      "Effect.Console"
    ]
  },
  "Data.Monoid.Endo": {
    "path": ".spago/prelude/v4.1.1/src/Data/Monoid/Endo.purs",
    "depends": [
      "Prelude",
      "Control.Applicative",
      "Control.Apply",
      "Data.Functor",
      "Data.Function",
      "Control.Category",
      "Control.Semigroupoid",
      "Data.Boolean",
      "Data.Ord",
      "Data.Eq",
      "Data.HeytingAlgebra",
      "Data.Symbol",
      "Data.Unit",
      "Data.Show",
      "Record.Unsafe",
      "Type.Data.RowList",
      "Type.Data.Row",
      "Data.Void",
      "Data.Ordering",
      "Data.Semigroup",
      "Data.Ring",
      "Data.Semiring",
      "Control.Bind",
      "Control.Monad",
      "Data.BooleanAlgebra",
      "Data.Bounded",
      "Data.CommutativeRing",
      "Data.DivisionRing",
      "Data.EuclideanRing",
      "Data.Field",
      "Data.Monoid",
      "Data.NaturalTransformation"
    ]
  },
  "Control.Applicative": {
    "path": ".spago/prelude/v4.1.1/src/Control/Applicative.purs",
    "depends": [
      "Control.Apply",
      "Data.Functor",
      "Data.Function",
      "Control.Category",
      "Control.Semigroupoid",
      "Data.Boolean",
      "Data.Ord",
      "Data.Eq",
      "Data.HeytingAlgebra",
      "Data.Symbol",
      "Data.Unit",
      "Data.Show",
      "Record.Unsafe",
      "Type.Data.RowList",
      "Type.Data.Row",
      "Data.Void",
      "Data.Ordering",
      "Data.Semigroup",
      "Data.Ring",
      "Data.Semiring"
    ]
  },
  "Record.Unsafe": {
    "path": ".spago/prelude/v4.1.1/src/Record/Unsafe.purs",
    "depends": []
  }
}
```

I think we should not have a separate command though, but run this check before performing any build, and fail the build if we're trying to import something from a "hidden package" (i.e. not specified in the dependencies of the project)

**Comment by marcosh** [comment]
Oh, that's nice!

If the `purs graph` command had explicitely also the information on the package (the information is actually there, but hidden in the path), then performing this check would become very straightforward.

I agree that running this check at every build it the optimal solution. It should probably fail if we're importing from a non-declared dependency, while it should provide a notice if a useless package is declared

**Comment by f-f** [comment]
This is not blocked anymore, since `purs graph` shipped in `v0.13.8` :blush:

**Comment by colinwahl** [comment]
I'm interested in working on this - I can put up a first pass at it soon.
